### PR TITLE
fix watchdog issue after coldboot (legacy)

### PR DESCRIPTION
**Pull request by piotr-kleins**
Watchdog stops working after coldboot. The fix is to move its configuration
from romstage to mainboard initialization, when MMIO is fully functional.
Hence WDT registers can be properly written.

Signed-off-by: Piotr Kleinschmidt <piotr.kleinschmidt@3mdeb.com>

### CODE REVIEW

**Comment by miczyg1** [comment]
@piotr-kleins we want the watchdog to be enabled as soon as possible. In mainline the watchdog is being enabled before AmdInitEarly: https://github.com/pcengines/coreboot/blob/develop/src/mainboard/pcengines/apu2/OemCustomize.c#L148
Have you checked whether watchdog works before AmdInitEarly in legacy?

**Comment by piotr-kleins** [comment]
@miczyg1 I have checked if moving watchdog configuration to `mb/pcengines/apu2/PlatformGnbPcie.c` and calling it inside `OemCustomizeInitEarly()` function works. It works, but there is no debug output available (I have added necessary includes, but no output at all). I thought it might be too much overhead to enable it, so I decided to move code to mainboard.c.

**Comment by miczyg1** [comment]
@piotr-kleins no, it is not an overhead. You simply need to either disable legacy free option or reenable decoding of LPC cycles for UART

File: src/mainboard/pcengines/apu2/mainboard.c [hunk 1]
```diff
@@ -133,6 +134,27 @@ static void mainboard_enable(device_t dev)
 		sio_dev = dev_find_slot_pnp(0x2E, NCT5104D_GPIO1);
 		if ( sio_dev ) sio_dev->enabled = 1;
 	}
+
```
Comment (miczyg1) [inline src/mainboard/pcengines/apu2/mainboard.c:137]: @piotr-kleins please do not leave commented code

Comment (piotr-kleins) [reply src/mainboard/pcengines/apu2/mainboard.c]: @miczyg1 Sorry, I was sure I delted it. Now it is.

**Pull request merged**